src/sipmpdf/functions.py

```diff
@@ -223,25 +237,56 @@ def darkcurrent_response(x: np.float64,
   eps = gain * resolution
   lo = pedestal
   up = pedestal + gain
+  
   return kern.where((x > (lo + eps)) & (x < (up - eps)),
-                    ((1 / (x - lo)) + (1 / (up - x))) / (2 * kern.log(
-                      (up - lo - eps) / eps)), 0)
+                    ((1 / (x - lo)) + (1 / (up - x))) / (2 * kern.log((up - lo - eps) / eps)), 0)
 
 
+##This part needs work, uncelar if it works correctly
 def ap_response_smeared(x: np.float64, smear: np.float64, n_ap: np.int64,
-                        beta: np.float64) -> np.float64:
-  kern = kernel_switch(x, smear, n_np, beta)
-  pass
-
+                        beta: np.float64,pedestal: np.float64, 
+                        gain: np.float64,total: np.int64) -> np.float64:
+  kern = kernel_switch(x, smear, n_ap, beta, pedestal, gain, total)
+  
+  ##did this based on the paper which says this only matters when i<2
+  if n_ap>1:
```

Comment:
Instead of a python `if` statement, you should use `kern.where` [1]

[1] https://numpy.org/doc/stable/reference/generated/numpy.where.html